core/services/llo/data_source.go

```diff
@@ -147,26 +150,36 @@ func (d *dataSource) Observe(ctx context.Context, streamValues llo.StreamValues,
 
 	wg.Wait()
 
+	elapsed := time.Since(now)
+
 	// Failed observations are always logged at warn level
+	var successfulStreamIDs []streams.StreamID
 	var failedStreamIDs []streams.StreamID
-	if len(errs) > 0 {
+	var errStrs []string
+
+	// Only log errors or if VerboseLogging is turned on
+	if len(errs) > 0 || opts.VerboseLogging() {
+		successfulStreamIDs = make([]streams.StreamID, 0, len(streamValues))
+		for strmID := range streamValues {
+			successfulStreamIDs = append(successfulStreamIDs, strmID)
+		}
```

Comment:
For comparison, with `iter.Seq` APIs, this could be written as:
```suggestion
	successfulStreamIDs = slices.AppendSeq(maps.Keys(streamValues))
```
Not clearly better though :shrug:

Reply:
This was actually broken anyway, rewritten and fixed